File: templates/python/code-carl/ch09_dp/t29_find_length_of_lcis.py
"""LeetCode 674. Longest Continuous Increasing Subsequence
https://leetcode.cn/problems/longest-continuous-increasing-subsequence/
"""

from typing import List


class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        pass

if __name__ == "__main__":
    solution = Solution()
    assert solution.findLengthOfLCIS([1, 3, 5, 4, 7]) == 3
    assert solution.findLengthOfLCIS([2, 2, 2, 2, 2]) == 1
    print("All tests passed.")

File: solutions/java/code-carl/ch09_dp/T29_FindLengthOfLcis.java
/**
 * Practice template for:
 * - LeetCode 674. Longest Continuous Increasing Subsequence
 *   https://leetcode.cn/problems/longest-continuous-increasing-subsequence/
 */
import java.util.Arrays;

/**
 * @author Cheng Yihao
 * @version 1.0
 * @date 2024/11/25 15:35
 * @comment
 */
public class T29_FindLengthOfLcis {
    public int findLengthOfLCIS(int[] nums) {
        int len = nums.length;
        int[] dp = new int[len];
        Arrays.fill(dp, 1);
        int maxLen = 1;
        for (int i = 1; i < len; i++) {
            if (nums[i] > nums[i - 1]) {
                dp[i] = dp[i - 1] + 1;
                maxLen = Math.max(maxLen, dp[i]);
            } else {
                dp[i] = 1;
            }
        }
        return maxLen;
    }

}
